`Vec<Vec<(String, usize)>>` panics with unsatisfied trait bounds

Hi everyone. Probably a stupid question but I'm getting this error when trying to return `Vec<Vec<(String, usize)>>` from a function and then define it as a singleton method:
```bash
error[E0599]: the method `call_handle_error` exists for fn item `fn(String, String) -> Vec<Vec<(String, usize)>> {reg}`, but its trait bounds were not satisfied
  --> ext/xre2/src/lib.rs:98:43
   |
98 |     module.define_singleton_method("reg", function!(reg, 2))?;
   |                                           ^^^^^^^^^^^^^^^^^ method cannot be called due to unsatisfied trait bounds
   |
   = note: the following trait bounds were not satisfied:
           `<fn(String, String) -> Vec<Vec<(String, usize)>> {reg} as FnOnce<(&Ruby, _, _)>>::Output = _`
           which is required by `fn(String, String) -> Vec<Vec<(String, usize)>> {reg}: RubyFunction2<_, _, _>`
           `fn(String, String) -> Vec<Vec<(String, usize)>> {reg}: Fn<(&Ruby, _, _)>`
           which is required by `fn(String, String) -> Vec<Vec<(String, usize)>> {reg}: RubyFunction2<_, _, _>`
           `<&fn(String, String) -> Vec<Vec<(String, usize)>> {reg} as FnOnce<(&Ruby, _, _)>>::Output = _`
           which is required by `&fn(String, String) -> Vec<Vec<(String, usize)>> {reg}: RubyFunction2<_, _, _>`
           `&fn(String, String) -> Vec<Vec<(String, usize)>> {reg}: Fn<(&Ruby, _, _)>`
           which is required by `&fn(String, String) -> Vec<Vec<(String, usize)>> {reg}: RubyFunction2<_, _, _>`
           `&mut fn(String, String) -> Vec<Vec<(String, usize)>> {reg}: Fn<(_, _)>`
           which is required by `&mut fn(String, String) -> Vec<Vec<(String, usize)>> {reg}: Function2<_, _, _>`
           `<&mut fn(String, String) -> Vec<Vec<(String, usize)>> {reg} as FnOnce<(&Ruby, _, _)>>::Output = _`
           which is required by `&mut fn(String, String) -> Vec<Vec<(String, usize)>> {reg}: RubyFunction2<_, _, _>`
           `&mut fn(String, String) -> Vec<Vec<(String, usize)>> {reg}: Fn<(&Ruby, _, _)>`
           which is required by `&mut fn(String, String) -> Vec<Vec<(String, usize)>> {reg}: RubyFunction2<_, _, _>`
   = note: this error originates in the macro `function` (in Nightly builds, run with -Z macro-backtrace for more info)
```

which in my very limited experience meant that there's no way to implicitly transform this rust type into ruby type(might be wrong, please correct me). So I was wondering why does `Vec<(String, usize)>` is ok but this one is not?

--- a/src/r_array.rs
+++ b/src/r_array.rs
@@ -1675,6 +1675,8 @@ where
     }
 }
 
+unsafe impl<T> IntoValueFromNative for Vec<T> where T: IntoValueFromNative {}
+
 #[cfg(feature = "old-api")]
 impl<T> FromIterator<T> for RArray
 where

